test/testing/test_waiting.py

```diff
@@ -36,7 +36,6 @@ def count_forever():
         def cb(args, kwargs, res, exc_info):
             if res == 10:
                 return True
-            return False
```

Comment:
Yes I wondered about this. I nearly did this, but then changed my mind (for a reason I can't recall and that now doesn't seem valid). I'll update them